### PR TITLE
Pde/multiple optimizely

### PR DESCRIPTION
**Pull request by jcger**
Trying to make it easier to have two optimizely client instances at the same time

### CODE REVIEW

**Comment by jcger** [comment]
missing tests and documentation

File: packages/sui-pde/README.md [hunk 1]
```diff
@@ -250,3 +250,40 @@ const MyComponent = () => {
 #### Force feature flag to be on/off
 
 It's slighty different to force a feature flag to be activated or deactivated. Lets assume we have our feature flag `ff_skills_field` running under `http://myweb.com`. In order to force the flag to be on or off you'll have to add a query param using the flag's name but adding `suipde_` as prefix same way we force an experiment, but the only valid values are on or off. For example, in this case, the url to open in order to force would be `http://myweb.com?suipde_ff_skills_field=on`. This would force the feature flag to be on. `http://myweb.com?suipde_ff_skills_field=off` would set the feature flag as off. If forced, optimizely impression will not be triggered.
+
+### Multiple Optimizely Adapters
+
+Meant to exist if you need more than one decision taking optimizely sdk.
+Integration with segment will only work for the first one so it will be necesary to configure segment so it forwards the events to the other optimizely destinations.
```
Comment (rmoralp) [inline packages/sui-pde/README.md:257]: What we should do in Segment config to enable it?

File: packages/sui-pde/src/adapters/optimizely/index.js [hunk 2]
```diff
@@ -53,14 +51,16 @@ export default class OptimizelyAdapter {
    * @param {object} param
    * @param {string} param.sdkKey
    * @param {object=} param.datafile
-   * @param {object} param.options https://docs.developers.optimizely.com/full-stack/docs/initialize-sdk-javascript-node
-   * @param {object} optimizely test purposes only, optimizely sdk
+   * @param {object} param.options datafile options https://docs.developers.optimizely.com/full-stack/docs/initialize-sdk-javascript-node
+   * @param {object} param.optimizely test purposes only, optimizely sdk
+   * @param {function} param.eventDispatcher https://docs.developers.optimizely.com/full-stack/docs/initialize-sdk-javascript-node
```
Comment (rmoralp) [inline packages/sui-pde/src/adapters/optimizely/index.js:56]:
```suggestion
   * @param {function} param.eventDispatcher https://docs.developers.optimizely.com/full-stack/docs/configure-event-dispatcher-javascript-node
```

File: packages/sui-pde/src/hooks/useExperiment.js [hunk 3]
```diff
@@ -8,14 +8,16 @@ import {getPlatformStrategy} from './common/platformStrategies.js'
...
+ * @param {string} param.adapterId Adapter id to be executed
  * @return {{variation: string}}
  */
 export default function useExperiment({
   experimentName,
   attributes,
   trackExperimentViewed,
-  queryString // for test purposes only
+  queryString,
+  adapterId
```
Comment (rmoralp) [inline packages/sui-pde/src/hooks/useExperiment.js:20]: Is `adapterId ` required?
Could you add some examples about useFeature and useExperiment when using multiple adapters?

File: packages/sui-pde/src/adapters/optimizely/multiple.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      })
+
+      return acc
+    }, {})
+  }
+
+  #adapters = {}
+
+  constructor(optimizelyAdapters) {
+    defaultAdapterId = Object.keys(optimizelyAdapters)[0] // first adapter will be the defaultAdapter
```
Comment (rmoralp) [inline packages/sui-pde/src/adapters/optimizely/multiple.js:32]: If `optimizelyAdapters` object keys are adapterId's, why not find for `default` one first?

**Review by rmoralp — COMMENTED** [review]
What do you think about cover it with some testing?

File: packages/sui-pde/src/adapters/optimizely/multiple.js (hunk 4 above)
Comment (alextremp) [reply packages/sui-pde/src/adapters/optimizely/multiple.js:32]: adapterIds are not required to have a `default` key, you could pass:
```
const multiple = new MultipleOptimizelyAdapter({
   adapter1: firstAdapter,
   adapter2: secondAdapter,
})
```

In this case, the `defaultAdapterId` would be `adapter1` so when using PDE without passing any `adapterId` in the calls, `adapter1` will be used.

File: packages/sui-pde/src/hooks/useExperiment.js (hunk 3 above)
Comment (alextremp) [reply packages/sui-pde/src/hooks/useExperiment.js:20]: done

**Pull request merged**